packages/camera/camera/ios/Classes/CameraPlugin.m

```diff
@@ -21,19 +21,31 @@ @interface FLTSavePhotoDelegate : NSObject <AVCapturePhotoCaptureDelegate>
 @end
 
 @interface FLTImageStreamHandler : NSObject <FlutterStreamHandler>
+@property(nonatomic, strong) dispatch_queue_t dispatchQueue;
 @property FlutterEventSink eventSink;
 @end
 
 @implementation FLTImageStreamHandler
 
+- (instancetype)initWithDispatchQueue:(dispatch_queue_t)dispatchQueue {
+  self = [super init];
+  NSAssert(self, @"super init cannot be nil");
+  _dispatchQueue = dispatchQueue;
+  return self;
+}
+
 - (FlutterError *_Nullable)onCancelWithArguments:(id _Nullable)arguments {
-  _eventSink = nil;
+  dispatch_async(self.dispatchQueue, ^{
+    self.eventSink = nil;
+  });
   return nil;
 }
 
 - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
                                        eventSink:(nonnull FlutterEventSink)events {
-  _eventSink = events;
+  dispatch_async(self.dispatchQueue, ^{
+    self.eventSink = events;
```

Comment:
event sink needs to be invoked on the main queue, but the pointer `_eventSink` itself should only be accessed on the session queue.